File: typescript/gift_certificate/getGiftCertificateByCode.ts
// I'm using the .js extension here so this file can be run stand-alone using node. Normally, there would be no extension.
import { giftCertificateApi } from '../api.js';
import { GetGiftCertificateByCodeRequest} from 'ultracart_rest_api_v2_typescript';


let code = 'NRQPHPCFVK'

// by_code does not take an expansion variable.  it will return the entire object by default.
const getGiftCertificateByCodeRequest: GetGiftCertificateByCodeRequest = { code: code};
let gcResponse = await giftCertificateApi.getGiftCertificateByCode(getGiftCertificateByCodeRequest);
let giftCertificate = gcResponse.gift_certificate;

console.log(giftCertificate);

File: python/gift_certificate/get_gift_certificate_by_code.py
# get a gift certificate by code.

from ultracart.apis import GiftCertificateApi
from ultracart.rest import ApiException
from pprint import pprint
from samples import api_client

api_instance = GiftCertificateApi(api_client())

try:

    code = 'NRQPHPCFVK'

    # by_code does not take an expansion variable.  it will return the entire object by default.
    gc_response = api_instance.get_gift_certificate_by_code(code)
    gift_certificate = gc_response.gift_certificate

    pprint(gift_certificate)

except ApiException as e:
    print("Exception when calling GiftCertificateApi->get_gift_certificate_by_code: %s\n" % e)
